File: languages/div.java
class DivError extends Exception {
    public DivError(String message) {
        super(message);
    }
}

public class div {
    public static double divide(double x, double y) throws DivError {
        if (y == 0.0) {
            throw new DivError("division by zero");
        }
        return x / y;
    }
    
    public static void main(String[] args) {
        double x = 1.0;
        double y = 0.0;
        
        try {
            double z = divide(x, y);
            System.out.println("The result is: " + z);
        } catch (DivError e) {
            System.err.println("Division failed: " + e.getMessage());
        }
    }
}

File: languages/div.py
class DivError(Exception):
    def __init__(self, message="division by zero"):
        super().__init__(message)

def div(x, y):
    if y == 0.0:
        raise DivError()
    return x / y

def main():
    x, y = 1.0, 0.0
    
    try:
        z = div(x, y)
        print(f"The result is: {z}")
    except DivError as e:
        print(f"Division failed: {e}", file=__import__('sys').stderr)

if __name__ == "__main__":
    main()